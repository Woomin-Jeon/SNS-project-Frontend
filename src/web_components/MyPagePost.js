import React from 'react';
import AddPost from '../components/posts/AddPost';
import ShowPostOthersPage from '../components/posts/ShowPost';

function MyPagePost({
  currentUserState,
  postState,
  setPostState,
  commentState,
  setCommentState,
  loginState,
  setLoginState,
}) {
  const { post } = postState;
  const { id } = currentUserState;

  return (
    <div className="mypage-post">
      <AddPost
        currentUserState={currentUserState}
        postState={postState}
        setPostState={setPostState}
      />
      <div>
        {post.filter((v) => v.id === id).map((p, index) => (
          <div key={index}>
            <ShowPostOthersPage
              postState={postState}
              setPostState={setPostState}
              currentUserState={currentUserState}
              commentState={commentState}
              setCommentState={setCommentState}
              p={p}
              index={index}
              loginState={loginState}
              setLoginState={setLoginState}
            />
          </div>
        ))}
      </div>
    </div>
  );
}

export default MyPagePost;
