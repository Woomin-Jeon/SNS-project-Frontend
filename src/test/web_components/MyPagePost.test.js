import MyPagePost from "../../web_components/MyPagePost";
import React from "react";
import {shallow} from 'enzyme';

describe('MyPagePost', () => {
  let loginState;
  let setLoginState;
  let currentUserState;
  let postState;
  let setPostState;
  let commentState;
  let setCommentState;

  beforeEach(() => {
    postState = { post: [] };
    currentUserState = { id: 'Random_ID' };
  });

  it('renders well', () => {
    const component = shallow(
      <MyPagePost
        currentUserState={currentUserState}
        postState={postState}
        setPostState={setPostState}
        commentState={commentState}
        setCommentState={setCommentState}
        loginState={loginState}
        setLoginState={setLoginState}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
