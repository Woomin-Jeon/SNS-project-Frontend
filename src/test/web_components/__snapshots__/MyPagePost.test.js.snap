// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPagePost renders well 1`] = `ShallowWrapper {}`;
